### PR TITLE
Throw exception of conditions for node remove are not met

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java [hunk 1]
```diff
@@ -468,24 +469,26 @@ private Node move(Node node, Node.State toState, Agent agent, Optional<String> r
...
+        if (nodeToRemove.state().equals(Node.State.dirty)) {
+            if (!(nodeToRemove.flavor().getType().equals(Flavor.Type.DOCKER_CONTAINER))) {
+                throw new IllegalArgumentException("Only docker nodes can be deleted from state dirty");
+            }
         }
 
-        try (Mutex lock = lock(nodeToRemove.get())) {
-            return db.removeNode(nodeToRemove.get().state(), hostname);
+        try (Mutex lock = lock(nodeToRemove)) {
+            db.removeNode(nodeToRemove.state(), hostname);
```
Comment (hmusum) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java]: MInor: Change signature of the method called to be void as well?

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java [hunk 2]
```diff
@@ -71,17 +72,18 @@ public void featureToggleDynamicAllocationTest() {
...
-
-            assertFalse(tester.nodeRepository().remove("host2"));
+        tester.addNode("id1", "host1", "default", NodeType.host);
+        tester.addNode("id2", "host2", "docker", NodeType.tenant);
+        tester.nodeRepository().setDirty("host2");
 
-            tester.curator().set(Path.fromString("/provision/v1/dynamicDockerAllocation"), new byte[0]);
-            assertTrue(tester.nodeRepository().remove("host2"));
-        } finally {
-            tester.curator().delete(Path.fromString("/provision/v1/dynamicDockerAllocation"));
```
Comment (hmusum) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java]: Was the delete in finally not needed?

**Review by hmusum — COMMENTED** [review]
Two minor comments

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java (hunk 2 above)
Comment (freva) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java]: The other tests don't use it, additionally this is the last thing that happens in the test, and a new `Curator` is created for next test as part of `@Before`, the `MockCurator` uses memory FS, so delete shouldn't be necessary.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java (hunk 1 above)
Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java]: Changed

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java (hunk 2 above)
Comment (hmusum) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/NodeRepositoryTest.java]: OK, good

**Comment by hmusum** [comment]
👍

**Pull request merged**